### ISSUE TITLE
Build-in string format 'date-time' shall support all formats

### ISSUE THREAD

**Issue by axhero** [lateapexearlyspeed/Lateapexearlyspeed.JsonSchema#45]
Version 1.1.8
DotNet 8

The json schema defines the [build-in format](https://json-schema.org/understanding-json-schema/reference/string#built-in-formats) 'date-time', by "Dates and times are represented in [RFC 3339, section 5.6](https://tools.ietf.org/html/rfc3339#section-5.6)."

The definition supports "time-offset" format with "Z" (UTC time)
The definition supports optional "time-secfrac" with one or more digits.

This xunit test shall finish successfully:
```
    [Fact]
    public void TestDateTime()
    {
        string jsonschema = """
         {
            "$schema": "https://json-schema.org/draft/2020-12/schema",
            "type": "object",

            "properties": {
                "DateTimeValue": {
                    "type": "string",
                    "format": "date-time"
                }
            }
         }
         """;

        var jv = new LateApexEarlySpeed.Json.Schema.JsonValidator(jsonschema);

        string jsondata = "{ \"DateTimeValue\": \"" + DateTime.Now.ToString("yyyy-MM-ddTHH:mm:ss+00:00", CultureInfo.InvariantCulture) + "\" }";
        Assert.True(jv.Validate(jsondata).IsValid);

        jsondata = "{ \"DateTimeValue\": \"" + DateTime.Now.ToString("yyyy-MM-ddTHH:mm:ss.fff+00:00", CultureInfo.InvariantCulture) + "\" }";
        Assert.True(jv.Validate(jsondata).IsValid);

        jsondata = "{ \"DateTimeValue\": \"" + DateTime.Now.ToString("yyyy-MM-ddTHH:mm:ss.fffffff+00:00", CultureInfo.InvariantCulture) + "\" }";
        Assert.True(jv.Validate(jsondata).IsValid);

        jsondata = "{ \"DateTimeValue\": \"" + DateTime.Now.ToString("yyyy-MM-ddTHH:mm:ssZ", CultureInfo.InvariantCulture) + "\" }";
        Assert.True(jv.Validate(jsondata).IsValid);

        jsondata = "{ \"DateTimeValue\": \"" + DateTime.Now.ToString("yyyy-MM-ddTHH:mm:ss.fffZ", CultureInfo.InvariantCulture) + "\" }";
        Assert.True(jv.Validate(jsondata).IsValid);

        jsondata = "{ \"DateTimeValue\": \"" + DateTime.Now.ToString("yyyy-MM-ddTHH:mm:ss.fffffffZ", CultureInfo.InvariantCulture) + "\" }";
        Assert.True(jv.Validate(jsondata).IsValid);

    }

```

**Comment by lateapexearlyspeed** [comment]
#42 also contains similar requirement, fixing.

**Comment by lateapexearlyspeed** [comment]
@axhero fix is in version 1.1.9.
